pkg/ccl/streamingccl/streamingest/stream_replication_e2e_test.go

```diff
@@ -995,3 +1007,131 @@ func TestTenantStreamingMultipleNodes(t *testing.T) {
+		for i, v := range insert {
+			c.srcTenantSQL.Exec(t, "INSERT INTO d.t (a, b) VALUES ($1, $2)", i+1000, v)
+		}
+
+		preImportData := c.srcTenantSQL.QueryStr(t, `SELECT * FROM d.t`)
+		timeBeforeImport := c.srcSysServer.Clock().Now()
+
+		// The import job hit an injected failure during import and got paused.
+		c.srcTenantSQL.ExpectErr(
+			t, `.*pausing due to error.*`,
+			`IMPORT INTO d.t (a, b) CSV DATA ($1)`, "nodelocal://0/data-0")
+		importJobID := jobutils.GetLastJobIDByType(t, c.srcTenantSQL, "IMPORT")
+		jobutils.WaitForJobToPause(t, c.srcTenantSQL, importJobID)
+
+		c.waitUntilHighWatermark(c.srcSysServer.Clock().Now(), jobspb.JobID(ingestionJobID))
+		jobutils.WaitForJobToPause(t, c.destTenantSQL, importJobID)
+
+		// Cancel the import job, the imported data got rolled back through a DelRange
+		c.srcTenantSQL.Exec(t, "CANCEL JOB $1", importJobID)
+		jobutils.WaitForJobToCancel(t, c.srcTenantSQL, importJobID)
+
+		c.srcTenantSQL.CheckQueryResults(t, `SELECT * FROM d.t`, preImportData)
+		c.waitUntilHighWatermark(c.srcSysServer.Clock().Now(), jobspb.JobID(ingestionJobID))
+		c.compareResult("SELECT * FROM d.t")
+		c.compareResult(fmt.Sprintf("SELECT * FROM d.t AS OF SYSTEM TIME %d", timeBeforeImport.WallTime))
+	})
+
+	t.Run("cancel-import", func(t *testing.T) {
+		defer gcjob.SetSmallMaxGCIntervalForTest()()
+		c.srcTenantSQL.Exec(t, "CREATE DATABASE failedimport; USE failedimport;")
+		// The import job hit an injected failure during import and got paused.
+		c.srcTenantSQL.ExpectErr(
+			t, `.*pausing due to error.*`,
+			fmt.Sprintf(`IMPORT TABLE simple FROM PGDUMP ('%s') WITH ignore_unsupported_statements`, "nodelocal://0/simple.sql"),
+		)
+		// Import job gets paused and cancel the job
+		importJobID := jobutils.GetLastJobIDByType(t, c.srcTenantSQL, "IMPORT")
+
+		jobutils.WaitForJobToPause(t, c.srcTenantSQL, importJobID)
+		c.srcTenantSQL.Exec(t, "CANCEL JOB $1", importJobID)
+		jobutils.WaitForJobToCancel(t, c.srcTenantSQL, importJobID)
+
+		gcJobID := jobutils.GetLastJobID(t, c.srcTenantSQL)
+		jobutils.WaitForJobToSucceed(t, c.srcTenantSQL, gcJobID)
+
+		c.waitUntilHighWatermark(c.srcSysServer.Clock().Now(), jobspb.JobID(ingestionJobID))
+		jobutils.WaitForJobToSucceed(t, c.destTenantSQL, gcJobID)
+	})
+
+	cmpExecCmp := func(t *testing.T, execStmts []string, cmpBefore []string, cmpAfter []string) {
+		c.t = t
+		if len(cmpBefore) > 0 {
+			c.waitUntilHighWatermark(c.srcSysServer.Clock().Now(), jobspb.JobID(ingestionJobID))
+		}
+		for _, cmp := range cmpBefore {
+			c.compareResult(cmp)
+		}
+
+		timeBeforeExec := c.srcSysServer.Clock().Now()
+		c.srcTenantSQL.ExecMultiple(c.t, execStmts...)
+
+		if len(cmpAfter) > 0 {
+			c.waitUntilHighWatermark(c.srcSysServer.Clock().Now(), jobspb.JobID(ingestionJobID))
+		}
+		for _, cmp := range cmpAfter {
+			c.compareResult(cmp)
+			if strings.Contains(cmp, "SELECT") {
+				c.compareResult(fmt.Sprintf("%s AS OF SYSTEM TIME %d", cmp, timeBeforeExec.WallTime))
+			}
+		}
```

Comment:
If you wanted to avoid running the destination tenant sql pod, you could also use the new `compareTenantFingerprintsAtTimestamp` once bors merges it.  But we can also do that in a follow up laster.